### PR TITLE
ci: Upload apisix-openresty/apisix/dashboard artifact

### PR DESCRIPTION
**Pull request by imjoey**
Changes proposed with this PR:
1. Upload the rpms of apisix-openresty, apisix that depends on apisix-openresty;
2. Upload the debs of apisix-openresty, apisix, dashboard;
3. Bump the versions of apisix, dashboard;

Signed-off-by: imjoey <majunjie@apache.org>

### CODE REVIEW

File: .github/workflows/package-apisix-dashboard-deb-ubuntu20.04.yml [hunk 1]
```diff
@@ -54,3 +54,9 @@ jobs:
               exit 1
           fi
 
+      - name: Publish Artifact
```
Comment (nic-chen) [inline .github/workflows/package-apisix-dashboard-deb-ubuntu20.04.yml:57]: we could add `retention-days` for it.

Comment (imjoey) [reply .github/workflows/package-apisix-dashboard-deb-ubuntu20.04.yml:57]: @nic-chen yep, 7 days is more proper. How about change this in another PR? :) Thanks.

Comment (nic-chen) [reply .github/workflows/package-apisix-dashboard-deb-ubuntu20.04.yml:57]: sounds good, we could add in another PR uniformly.

**Review by nic-chen — APPROVED** [review]

File: .github/workflows/package-apisix-rpm-el7-with-apisix-openresty.yml [hunk 2]
```diff
@@ -37,6 +37,7 @@ jobs:
       - name: packaging APISIX
         run: |
           make package type=rpm app=apisix version=2.7 checkout=release/2.7 image_base=centos image_tag=7 openresty=apisix-openresty
+          make package type=rpm app=apisix version=2.9 checkout=release/2.9 image_base=centos image_tag=7 openresty=apisix-openresty
```
Comment (spacewander) [inline .github/workflows/package-apisix-rpm-el7-with-apisix-openresty.yml:40]: Why use 2.9?

Comment (imjoey) [reply .github/workflows/package-apisix-rpm-el7-with-apisix-openresty.yml:40]: @spacewander in order to put the rpm of the latest apisix to yum repository. What do you think? Thanks.

Comment (imjoey) [reply .github/workflows/package-apisix-rpm-el7-with-apisix-openresty.yml]: @spacewander  After reconsidering this, I prefer to discard the adding of `2.9`. PR updated. Thanks for reminding.

**Review by tzssangglass — APPROVED** [review]

**Pull request merged**